trollflow2/plugins/__init__.py

```diff
@@ -1114,3 +1114,29 @@ def callback_close(obj, targs, job, fmat_config):
         for targ in targs:
             targ.close()
     return obj
+
+
+def use_fsspec_cache(job):
+    """Use the caching from fsspec for (remote) files."""
+    import fsspec
+    from satpy.readers import FSFile
+
+    cache = job["product_list"]["fsspec_cache"]["type"]
+    cache_options = job["product_list"]["fsspec_cache"].get("options")
+    filenames = job["input_filenames"]
+    cached_filenames = [f"{cache}::{f}" for f in filenames]
+    kwargs = {}
+    if cache_options:
+        kwargs[cache] = cache_options
+    open_files = fsspec.open_files(cached_filenames, **kwargs)
+    fs_files = [FSFile(open_file) for open_file in open_files]
+    job["input_filenames"] = fs_files
+
+
+def clear_fsspec_cache(job):
+    """Clear all files in fsspec cache directory."""
+    filenames = job["input_filenames"]
+
+    for f in filenames:
+        if hasattr(f, "_fs"):
+            f._fs.clear_cache()
```

Comment:
It's not so nice that we need to access a private attribute of FSFile, should we add a public `fs` property (read-only) to FSFile instead?

Reply:
Yeah, that would be great. I'll have a go tomorrow if I remember.

Reply:
Did this already today: https://github.com/pytroll/satpy/pull/2832

Reply:
I've now updated this PR to use the public `.fs` property to clean the cache. The tests will fail until a new Satpy release is available.